fsspec/caching.py

```diff
@@ -422,6 +422,30 @@ def _fetch(self, start, end):
         return self.data[start:end]
 
 
+class KnownPartsOfAFile(BaseCache):
+    name = "parts"
+
+    def __init__(self, blocksize, fetcher, size, data={}, **_):
```

Comment:
It looks like you are assuming that the user has already read the necessary byte ranges from the file and has used them to populate the values of `data` before the file is open and this cache is initialized. Am I mistaken?

To clarify, I want to make sure that we can specify only the known byte ranges (not the actual data stored at those byte ranges) when we call `fs.open`.  I would expect the necessary byte ranges to be fetched automatically from the file (using cat_ranges) when this cache is initialized.  Does that make sense, or am I misunderstanding the API?